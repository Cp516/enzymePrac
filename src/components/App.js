import React from 'react';
import Square from './Square'
import './styles/App.css';


class App extends React.Component {
  constructor(){
    super();
    this.state = {
      board : ['','','','','','','','',''],
      value: 'X',
      isWinner: false,
      winner: ''
    }
  }
  squareClickHandler = (loc) =>{
    const transitionBoard = this.state.board;
    transitionBoard.splice(loc, 1, this.state.value) 
    this.setState({board: transitionBoard, value: this.state.value === 'X' ? 'O' : 'X'})

  }

  render() {
    return (
      <div>
          {/* <Square  loc={0} squareClickHanlder={this.squareClickHandler} value={this.state.value}/> */}
        
        <div className='board'>
          {this.state.board.map((square, i) => {return <Square key={i} loc={i} squareClickHandler={this.squareClickHandler} value={this.state.value}/>})}
        </div>
      </div>
    );
  }
}

export default App;
