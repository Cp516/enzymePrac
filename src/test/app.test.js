import React from 'react'
import { shallow } from 'enzyme'
import testSetup from './../testSetup'
import App from './../components/App'

describe('App', ()=>{
    const app = shallow(<App />);
    const boardArr = ['','','','','','','','','',];
    const boardArr1 = ['X','','','','','','','','',];

    it('should render properly', ()=>{
        expect(app).toMatchSnapshot()
    });
    
    it('App `state` should contain `board`', ()=>{
        expect(app.state().board).toEqual(boardArr)
    });

    // How specific should tests be should they separated or together like this
    // When do we start a new describe 

    it('App `state` should contain `isWinner, winner` keys', ()=>{
        expect(app.state().isWinner).toEqual(false);
    });

    it('App `state` should contain `winner` keys', ()=>{
        expect(app.state().winner).toEqual('');
    });

    it('App `state` should contain `value` keys', ()=>{
        expect(app.state().value).toEqual('X');
    });

    it('App should contain a `div` with a className board', ()=>{
        expect(app.find('.board').exists()).toEqual(true)
    });

    it('App should contain a component called Square', ()=>{
        expect(app.find('Square').exists()).toEqual(true)
    });

    it('App should contain a `div` with a className board with length of 9', ()=>{
        expect(app.find('.board').children().length).toEqual(9)
    });
    //What exactly do backticks signify and when should they used
    describe('`squareClickHandler` should update `board`',()=>{
        beforeEach(()=>{
            app.instance().squareClickHandler('X')
        });

        afterEach(()=>{
            app.setState({board: boardArr, value:'X'})
        });

        it('squareClickHandler Fn call should update `board` in `state`', ()=>{
            expect(app.state().board).toEqual(boardArr1)
        });
        
        it('squareClickHandler should initially update `value` from "X" to "O"',()=>{
            expect(app.state().value).toEqual('O')
        });
    });
    
});

// 2.) When do we start a new describe, what exactly do they signify

// 3.) What exactly do back ticks signify and when should they used