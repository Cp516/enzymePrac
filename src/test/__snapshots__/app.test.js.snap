// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="board"
>
        <Square
                loc={0}
                squareClickHandler={[Function]}
                value="X"
        />
        <Square
                loc={1}
                squareClickHandler={[Function]}
                value="X"
        />
        <Square
                loc={2}
                squareClickHandler={[Function]}
                value="X"
        />
        <Square
                loc={3}
                squareClickHandler={[Function]}
                value="X"
        />
        <Square
                loc={4}
                squareClickHandler={[Function]}
                value="X"
        />
        <Square
                loc={5}
                squareClickHandler={[Function]}
                value="X"
        />
        <Square
                loc={6}
                squareClickHandler={[Function]}
                value="X"
        />
        <Square
                loc={7}
                squareClickHandler={[Function]}
                value="X"
        />
        <Square
                loc={8}
                squareClickHandler={[Function]}
                value="X"
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Square
            loc={0}
            squareClickHandler={[Function]}
            value="X"
/>,
          <Square
            loc={1}
            squareClickHandler={[Function]}
            value="X"
/>,
          <Square
            loc={2}
            squareClickHandler={[Function]}
            value="X"
/>,
          <Square
            loc={3}
            squareClickHandler={[Function]}
            value="X"
/>,
          <Square
            loc={4}
            squareClickHandler={[Function]}
            value="X"
/>,
          <Square
            loc={5}
            squareClickHandler={[Function]}
            value="X"
/>,
          <Square
            loc={6}
            squareClickHandler={[Function]}
            value="X"
/>,
          <Square
            loc={7}
            squareClickHandler={[Function]}
            value="X"
/>,
          <Square
            loc={8}
            squareClickHandler={[Function]}
            value="X"
/>,
        ],
        "className": "board",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "loc": 0,
            "squareClickHandler": [Function],
            "value": "X",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "loc": 1,
            "squareClickHandler": [Function],
            "value": "X",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "loc": 2,
            "squareClickHandler": [Function],
            "value": "X",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "loc": 3,
            "squareClickHandler": [Function],
            "value": "X",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "loc": 4,
            "squareClickHandler": [Function],
            "value": "X",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "loc": 5,
            "squareClickHandler": [Function],
            "value": "X",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "loc": 6,
            "squareClickHandler": [Function],
            "value": "X",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "function",
          "props": Object {
            "loc": 7,
            "squareClickHandler": [Function],
            "value": "X",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "function",
          "props": Object {
            "loc": 8,
            "squareClickHandler": [Function],
            "value": "X",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="board"
>
          <Square
                    loc={0}
                    squareClickHandler={[Function]}
                    value="X"
          />
          <Square
                    loc={1}
                    squareClickHandler={[Function]}
                    value="X"
          />
          <Square
                    loc={2}
                    squareClickHandler={[Function]}
                    value="X"
          />
          <Square
                    loc={3}
                    squareClickHandler={[Function]}
                    value="X"
          />
          <Square
                    loc={4}
                    squareClickHandler={[Function]}
                    value="X"
          />
          <Square
                    loc={5}
                    squareClickHandler={[Function]}
                    value="X"
          />
          <Square
                    loc={6}
                    squareClickHandler={[Function]}
                    value="X"
          />
          <Square
                    loc={7}
                    squareClickHandler={[Function]}
                    value="X"
          />
          <Square
                    loc={8}
                    squareClickHandler={[Function]}
                    value="X"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Square
              loc={0}
              squareClickHandler={[Function]}
              value="X"
/>,
            <Square
              loc={1}
              squareClickHandler={[Function]}
              value="X"
/>,
            <Square
              loc={2}
              squareClickHandler={[Function]}
              value="X"
/>,
            <Square
              loc={3}
              squareClickHandler={[Function]}
              value="X"
/>,
            <Square
              loc={4}
              squareClickHandler={[Function]}
              value="X"
/>,
            <Square
              loc={5}
              squareClickHandler={[Function]}
              value="X"
/>,
            <Square
              loc={6}
              squareClickHandler={[Function]}
              value="X"
/>,
            <Square
              loc={7}
              squareClickHandler={[Function]}
              value="X"
/>,
            <Square
              loc={8}
              squareClickHandler={[Function]}
              value="X"
/>,
          ],
          "className": "board",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "loc": 0,
              "squareClickHandler": [Function],
              "value": "X",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "loc": 1,
              "squareClickHandler": [Function],
              "value": "X",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "loc": 2,
              "squareClickHandler": [Function],
              "value": "X",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "loc": 3,
              "squareClickHandler": [Function],
              "value": "X",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "function",
            "props": Object {
              "loc": 4,
              "squareClickHandler": [Function],
              "value": "X",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "function",
            "props": Object {
              "loc": 5,
              "squareClickHandler": [Function],
              "value": "X",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "function",
            "props": Object {
              "loc": 6,
              "squareClickHandler": [Function],
              "value": "X",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "7",
            "nodeType": "function",
            "props": Object {
              "loc": 7,
              "squareClickHandler": [Function],
              "value": "X",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "8",
            "nodeType": "function",
            "props": Object {
              "loc": 8,
              "squareClickHandler": [Function],
              "value": "X",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
